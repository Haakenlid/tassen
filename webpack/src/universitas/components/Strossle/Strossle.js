const StrossleV1 = ({ id, url }) => (
  <div className="StrossleV1" data-spklw-widget={id} data-spklw-url={url} />
)

const StrossleV2 = ({ id, url }) => {
  const className = `strossle-widget-${id}`
  React.useEffect(
    () => {
      window.strossle && window.strossle(id, `.${className}`)
    },
    [url],
  )
  return <div className={className} />
}

const StrossleWidget = ({ v1 = false, ...props }) =>
  v1 ? <StrossleV1 {...props} /> : <StrossleV2 {...props} />

export default StrossleWidget
