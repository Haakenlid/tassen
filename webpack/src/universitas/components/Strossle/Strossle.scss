@import 'universitas/styles/_common.scss';

// Strossle widget V1 breaks layout on small screens
.StrossleV1 {
  @media (max-width: $content-width) {
    display: none;
  }
}
