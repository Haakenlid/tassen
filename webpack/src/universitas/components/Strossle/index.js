import Strossle from './Strossle.js'
import './Strossle.scss'
export default Strossle
