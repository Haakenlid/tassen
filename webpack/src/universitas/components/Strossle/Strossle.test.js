import Strossle from './index'
import renderer from 'react-test-renderer'

describe('Strossle', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<Strossle />).toJSON()
    expect(tree).toEqual(expect.any(Object))
    expect(tree).toMatchSnapshot()
  })
})
