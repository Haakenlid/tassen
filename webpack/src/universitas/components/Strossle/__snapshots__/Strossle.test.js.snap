// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Strossle renders correctly 1`] = `
<div
  className="strossle-widget-undefined"
/>
`;
